From bc3a5d9c5cffbc49f32c9e9eed2a3e582496275c Mon Sep 17 00:00:00 2001
Message-Id: <bc3a5d9c5cffbc49f32c9e9eed2a3e582496275c.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Sat, 27 Jul 2013 17:14:08 +0530
Subject: [PATCH 037/312] FOR_UPSTREAM [VPG]: drm/i915: Fix the panel vdd
 on/off flow

Panel VDD On/Off flow issue not really right for BYT.
This should now be fine and working on both FFRD and RVP

(cherry picked from commit 67bef4bc90f0265276cd6893e8f2dd452fb3f8d3)

Issue: APDEV-1168
Change-Id: I969a273e3a71df4a230a0a78a2c239dab9cbbab9
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
Reviewed-on: http://android.intel.com:8080/121446
Reviewed-by: Purushothaman, Vijay A <vijay.a.purushothaman@intel.com>
Reviewed-by: Wood, Brian J <brian.j.wood@intel.com>
Tested-by: Sun, Hang L <hang.l.sun@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Signed-off-by: A.Sunil Kamath <sunil.kamath@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 98e26de..6dbaa21 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -1169,7 +1169,7 @@ static bool _edp_panel_vdd_on(struct intel_dp *intel_dp)
 	 * If the panel wasn't on, delay before accessing aux channel
 	 */
 	if (!edp_have_panel_power(intel_dp)) {
-		DRM_DEBUG_KMS("eDP was not running\n");
+		DRM_DEBUG_KMS("eDP was not running; waiting for VDD on\n");
 		msleep(intel_dp->panel_power_up_delay);
 	}
 
@@ -1321,7 +1321,7 @@ void intel_edp_panel_off(struct intel_dp *intel_dp)
 	WARN(!intel_dp->want_panel_vdd, "Need VDD to turn off panel\n");
 
 	pp = ironlake_get_pp_control(intel_dp);
-	/* We need to switch off panel power _and_ force vdd, for otherwise some
+	/* We need to switch off panel power, for otherwise some
 	 * panels get very unhappy and cease to work. */
 	pp &= ~(POWER_TARGET_ON | PANEL_POWER_RESET | EDP_FORCE_VDD |
 		EDP_BLC_ENABLE);
@@ -1880,7 +1880,7 @@ static void intel_disable_dp(struct intel_encoder *encoder)
 
 	/* Some of the FFRD10 PR1.1B boards doesnt like when edp panel power
 	 * is off */
-	/* intel_edp_panel_off(intel_dp); */
+	intel_edp_panel_off(intel_dp);
 
 	/* cpu edp my only be disable _after_ the cpu pipe/plane is disabled. */
 	if (!(port == PORT_A || IS_VALLEYVIEW(dev)))
-- 
1.7.9.5

